sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveDefaultColumns.scala

```diff
@@ -569,25 +573,43 @@ case class ResolveDefaultColumns(catalog: SessionCatalog) extends Rule[LogicalPl
     // found" error. In the latter cases, return out of this rule without changing anything and let
     // the analyzer return a proper error message elsewhere.
     val tableName: TableIdentifier = source match {
-      case Some(r: UnresolvedRelation) => TableIdentifier(r.name)
-      case Some(r: UnresolvedCatalogRelation) => r.tableMeta.identifier
-      case _ => return None
+      case Some(r: UnresolvedRelation) =>
+        r.multipartIdentifier match {
```

Comment:
Shall we just call the method `ResolveRelations.resolveRelation`?

Reply:
Sure, good idea. This is done. Now we can use the relation cache therein to avoid looking up into the catalog twice, and also unify the logic to resolve relations.